### ISSUE TITLE
library builds but tests fail with curl 7.62 on master

### ISSUE THREAD

**Issue by boutil** [taf2/curb#369]
Hi,

https://github.com/taf2/curb/commit/aa23bf67b0290882514385ad25659d4511c882d2 fixes building curb against 7.62. However running tests fail with the following errors:
```
Loaded suite /usr/lib/ruby/vendor_ruby/rake/rake_test_loader
Started
.........F
===============================================================================================
     41:     assert_instance_of Curl::Easy, c = Curl::Easy.perform($TEST_URL)
     42:     assert_match(/^# DO NOT REMOVE THIS COMMENT/, c.body_str)
     43:     assert_match(/^# DO NOT REMOVE THIS COMMENT/, c.body)
  => 44:     assert_equal "", c.header_str
     45:   end    
     46: 
     47:   def test_class_perform_02
/home/boutil/src/curb/tests/tc_curl_easy.rb:44:in `test_class_perform_01'
<"">(UTF-8) expected but was
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">(ASCII-8BIT)

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
  
? Encoding: UTF  -8   
?           ASCII  BIT
Failure: test_class_perform_01(TestCurbCurlEasy)
===============================================================================================
F
===============================================================================================
     49:     assert_instance_of Curl::Easy, c = Curl::Easy.perform($TEST_URL) { |curl| curl.on_body { |d| data << d; d.length } }    
     50: 
     51:     assert_nil c.body_str
  => 52:     assert_equal "", c.header_str
     53:     assert_match(/^# DO NOT REMOVE THIS COMMENT/, data)
     54:   end    
     55: 
/home/boutil/src/curb/tests/tc_curl_easy.rb:52:in `test_class_perform_02'
<"">(UTF-8) expected but was
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">(ASCII-8BIT)

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
  
? Encoding: UTF  -8   
?           ASCII  BIT
Failure: test_class_perform_02(TestCurbCurlEasy)
===============================================================================================
.........................F
===============================================================================================
     144:     c = Curl::Easy.new($TEST_URL)    
     145:     assert_equal true, c.http_get
     146:     assert_match(/^# DO NOT REMOVE THIS COMMENT/, c.body_str)
  => 147:     assert_equal "", c.header_str
     148:   end    
     149: 
     150:   def test_get_02
/home/boutil/src/curb/tests/tc_curl_easy.rb:147:in `test_get_01'
<"">(UTF-8) expected but was
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">(ASCII-8BIT)

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
  
? Encoding: UTF  -8   
?           ASCII  BIT
Failure: test_get_01(TestCurbCurlEasy)
===============================================================================================
F
===============================================================================================
     156:     assert_equal true, c.http_get    
     157:     
     158:     assert_nil c.body_str
  => 159:     assert_equal "", c.header_str
     160:     assert_match(/^# DO NOT REMOVE THIS COMMENT/, data)
     161:   end    
     162: 
/home/boutil/src/curb/tests/tc_curl_easy.rb:159:in `test_get_02'
<"">(UTF-8) expected but was
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">(ASCII-8BIT)

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
  
? Encoding: UTF  -8   
?           ASCII  BIT
Failure: test_get_02(TestCurbCurlEasy)
===============================================================================================
................................................................#<Thread:0x00005582e80b4120@/home/boutil/src/curb/tests/tc_curl_easy.rb:26 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        6: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `block (2 levels) in test_threads'
        5: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `times'
        4: from /home/boutil/src/curb/tests/tc_curl_easy.rb:31:in `block (3 levels) in test_threads'
        3: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:240:in `assert_equal'
        2: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:53:in `assert_block'
        1: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:1636:in `_wrap_assertion'
/usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:55:in `block in assert_block': <"">("UTF-8") expected but was (Test::Unit::AssertionFailedError)
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">("ASCII-8BIT").

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
#<Thread:0x00005582e80b4648@/home/boutil/src/curb/tests/tc_curl_easy.rb:26 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        6: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `block (2 levels) in test_threads'
        5: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `times'
        4: from /home/boutil/src/curb/tests/tc_curl_easy.rb:31:in `block (3 levels) in test_threads'
        3: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:240:in `assert_equal'
        2: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:53:in `assert_block'
        1: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:1636:in `_wrap_assertion'
/usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:55:in `block in assert_block': <"">("UTF-8") expected but was (Test::Unit::AssertionFailedError)
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">("ASCII-8BIT").

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
#<Thread:0x00005582e80b44e0@/home/boutil/src/curb/tests/tc_curl_easy.rb:26 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        6: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `block (2 levels) in test_threads'
        5: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `times'
        4: from /home/boutil/src/curb/tests/tc_curl_easy.rb:31:in `block (3 levels) in test_threads'
        3: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:240:in `assert_equal'
        2: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:53:in `assert_block'
        1: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:1636:in `_wrap_assertion'
/usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:55:in `block in assert_block': <"">("UTF-8") expected but was (Test::Unit::AssertionFailedError)
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">("ASCII-8BIT").

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
#<Thread:0x00005582e80b4328@/home/boutil/src/curb/tests/tc_curl_easy.rb:26 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        6: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `block (2 levels) in test_threads'
        5: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `times'
        4: from /home/boutil/src/curb/tests/tc_curl_easy.rb:31:in `block (3 levels) in test_threads'
        3: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:240:in `assert_equal'
        2: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:53:in `assert_block'
        1: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:1636:in `_wrap_assertion'
/usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:55:in `block in assert_block': <"">("UTF-8") expected but was (Test::Unit::AssertionFailedError)
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">("ASCII-8BIT").

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
#<Thread:0x00005582e80b4878@/home/boutil/src/curb/tests/tc_curl_easy.rb:26 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        6: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `block (2 levels) in test_threads'
        5: from /home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `times'
        4: from /home/boutil/src/curb/tests/tc_curl_easy.rb:31:in `block (3 levels) in test_threads'
        3: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:240:in `assert_equal'
        2: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:53:in `assert_block'
        1: from /usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:1636:in `_wrap_assertion'
/usr/lib/ruby/vendor_ruby/test/unit/assertions.rb:55:in `block in assert_block': <"">("UTF-8") expected but was (Test::Unit::AssertionFailedError)
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">("ASCII-8BIT").

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
F
===============================================================================================
/home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `block (2 levels) in test_threads'
/home/boutil/src/curb/tests/tc_curl_easy.rb:27:in `times'
     28:           c = Curl.get($TEST_URL)
     29:           assert_match(/^# DO NOT REMOVE THIS COMMENT/, c.body_str)
     30:           assert_match(/^# DO NOT REMOVE THIS COMMENT/, c.body)
  => 31:           assert_equal "", c.header_str
     32:           assert_equal "", c.head
     33:         end
     34:       end
/home/boutil/src/curb/tests/tc_curl_easy.rb:31:in `block (3 levels) in test_threads'
<"">(UTF-8) expected but was
<"Content-Length: 5439\r\n" +
"Accept-ranges: bytes\r\n" +
"Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT\r\n" +
"\r\n">(ASCII-8BIT)

diff:
+ Content-Length: 5439
+ Accept-ranges: bytes
+ Last-Modified: Wed, 21 Nov 2018 08:50:56 GMT
  
? Encoding: UTF  -8   
?           ASCII  BIT
Failure: test_threads(TestCurbCurlEasy)
===============================================================================================
..........................................................
Finished in 11.593404388 seconds.
-----------------------------------------------------------------------------------------------
161 tests, 566 assertions, 5 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
96.8944% passed
-----------------------------------------------------------------------------------------------
13.89 tests/s, 48.82 assertions/s
rake aborted!
Command failed with status (1)
```

**Comment by taf2** [comment]
It looks like this might a change in how libcurl handles headers when dealing with file:// protocol?

**Comment by taf2** [comment]
I could be mistaken about this but since the test is just loading a file via file:// protocol it appears asserting the header_str is "" is not necessarily a good hard requirement so removing that assertion and keeping the important assertions for body is more important.

**Comment by taf2** [comment]
should be fixed now in 55c7f485672465461bb8b8a4589601c208a3d82c
